src/Microsoft.Azure.ServiceBus/Extensions/MessageInterOpExtensions.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.InteropExtensions
+{
+    using System;
+    using System.IO;
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// A Message Extension Class that provides extension methods to deserialize 
+    /// the body of a message that was serialized and sent to ServiceBus Queue/Topic
+    /// using the WindowsAzure.Messaging client library. The WindowsAzure.Messaging
+    /// client libary serializes objects using the 
+    /// <see cref="DataContractBinarySerializer"/> (default serializer) or <see cref="DataContractSerializer"/>
+    /// when sending message. This class provides extension methods to deserialize
+    /// and retrieve the body of such messages.
+    /// </summary>
+    /// <remarks>
+    /// 1. If a message is only being sent and received using this Microsoft.Azure.ServiceBus
+    /// client library, then the below extension methods are not relevant and should not be used.
+    /// 
+    /// 2. If this client library will be used to receive messages that were sent using both 
+    /// WindowsAzure.Messaging client library and this (Microsoft.Azure.ServiceBus) library,
+    /// then the Users need to add a User property <see cref="Message.UserProperties"/>
+    /// while sending the message. On receiving the message, this property can be examined to
+    /// determine if the message was from WindowsAzure.Messaging client library and if so
+    /// use the message.GetBody() extension method to get the actual body associated with the message.
+    /// 
+    /// ----------------------------------------------
+    /// Scenarios to use the GetBody Extension method:
+    /// ----------------------------------------------
+    /// If message was constructed using the WindowsAzure.Messaging client library as follows:
+    ///     var message1 = new BrokeredMessage("contoso"); // Sending a plain string
+    ///     var message2 = new BrokeredMessage(sampleObject); // Sending an actual customer object
+    ///     var message3 = new BrokeredMessage(Encoding.UTF8.GetBytes("contoso")); // Sending a UTF8 encoded byte array object
+    ///     
+    ///     await messageSender.SendAsync(message1);
+    ///     await messageSender.SendAsync(message2);
+    ///     await messageSender.SendAsync(message3);
+    ///     
+    /// Then retreive the original objects using this client library as follows:
+    /// (Serializer passed for GetBody() call can be null if message was sent using WindowsAzure.Messaging client
+    /// library and TransportType was AMQP.)
+    /// 
+    ///     var message1 = await messageReceiver.ReceiveAsync();
+    ///     var serializer1 = new DataContractBinarySerializer(typeof(string));
```

Comment:
This is a good question. We shall discuss further. For short term, we don't want to limit or provide serialization. The code here is only to support reading messages that were sent from old library. However, future is open to discuss.